chef/automate
# Vivek shankar/fixed 50k node error issue5278

### :nut_and_bolt: Description: What code changed, and why?


On sending nodes more than 50k, earlier data feed service used to throw error saying 4 mb grpc limit exceeded. This used to happen due to communication with cereal via grpc which collecting nodeIds. And the total size of 50k nodeIds used to become greater than 4mb
We implemented a chunk based solution to send data more than 4mb. 

### :chains: Related Resources
https://github.com/chef/automate/issues/5278

### :+1: Definition of Done

Automate data-feed should be able to handle more than 50k nodes and should not give the 4mb limit exceeded error. Data should be sent to configured data-feed integrations properly.
### :athletic_shoe: How to Build and Test the Change

- `rebuild component/cereal-service`
- `rebuild component/data-feed-service`
- Use the repository [automate-load-test-utils](https://github.com/chef/automate-load-test-utils/tree/master/automate-load-test) to load 50k plus data into automate
- Patch automate data-feed-service with an optimum interval required to send 50k data. eg- 4hr, 6hr
- Make sure to add a data-feed destination
- When the interval time is hit automate should be able to send all the 50k data

### :white_check_mark: Checklist

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests) - not required
- [ ] Docs added/updated? (all customer-facing changes) - not required

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
